### PR TITLE
sqlstring: Fix types of arguments

### PR DESCRIPTION
**Pull request by koooge**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mysqljs/sqlstring/blob/master/test/unit/test-SqlString.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## sqlstring ([<kbd>unpkg</kbd>](https://unpkg.com/browse/sqlstring@latest/))
was missing the following properties:
1. arrayToList
1. dateToString
1. bufferToString
1. objectToValues
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c84d73f2cbcedbf478c9f42f45e4e91f08ec5fc3
</p>

File: types/sqlstring/sqlstring-tests.ts [hunk 1]
```diff
@@ -32,3 +33,172 @@ const sql7 = SqlString.format("SELECT * FROM ?? WHERE ?? = ?", inserts);
...
+//   const buffer = new Buffer([0, 1, 254, 255]);
+//   const string = SqlString.escape(buffer);
+//   assert.strictEqual(string, "X'0001feff'");
+// }
+// {
+//   const buffer = new Buffer([0, 1, 254, 255]);
+//   buffer.toString = function() { return "00' OR '1'='1"; };
+//   const string = SqlString.escape(buffer);
+//   assert.strictEqual(string, "X'00\\' OR \\'1\\'=\\'1'");
+// }
```
Comment (koooge) [inline types/sqlstring/sqlstring-tests.ts:143]: Buffer requires node type reference. But it does not seem to good from the view of the performance.

Comment (amcasey) [reply types/sqlstring/sqlstring-tests.ts:143]: While I think you could reasonably add a type reference to the test file, you appear to have pretty good test coverage already, so I wouldn't worry about it.

**Review by amcasey — APPROVED** [review]

**Review by marvinhagemeister — APPROVED** [review]

**Comment by koooge** [comment]
Ready to merge

**Pull request merged**